### PR TITLE
Rename `reset_on_epoch` to `reset_on_run`

### PR DESCRIPTION
**Pull request by carmocca**
## What does this PR do?

And use the shortcuts where available

The `on_epoch` suffix doesn't apply to all progress objects.

### Does your PR introduce any breaking changes? If yes, please list them.

The method has been renamed

## Before submitting

- [ ] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [x] Did you make sure to **update the documentation** with your changes? (if necessary)
- [x] Did you write any **new necessary tests**? (not for typos and docs)
- [x] Did you verify new and **existing tests pass** locally with your changes?
- [x] Did you list all the **breaking changes** introduced by this pull request?
- [x] Did you **update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)**? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

### CODE REVIEW

**Review by tchaton — APPROVED** [review]
LGTM !

**Review by awaelchli — APPROVED** [review]

**Review by rohitgr7 — APPROVED** [review]

File: pytorch_lightning/trainer/progress.py [hunk 1]
```diff
@@ -151,6 +151,9 @@ def from_defaults(cls, tracker_cls: Type[ReadyCompletedTracker], **kwargs: int)
     def reset_on_epoch(self) -> None:
         self.current.reset()
 
+    def reset_on_run(self) -> None:
```
Comment (Borda) [inline pytorch_lightning/trainer/progress.py:154]: seems like deprecation public API, shall we make it back compatible?

File: pytorch_lightning/trainer/progress.py [hunk 2]
```diff
@@ -188,8 +191,8 @@ class BatchProgress(Progress):
 
     is_last_batch: bool = False
 
-    def reset_on_epoch(self) -> None:
-        super().reset_on_epoch()
+    def reset_on_run(self) -> None:
```
Comment (Borda) [inline pytorch_lightning/trainer/progress.py:194]: let's add a docstring to clarify if the reset is before a run or when...
how about "reset_before_run"?

File: pytorch_lightning/trainer/progress.py (hunk 1 above)
Comment (carmocca) [reply pytorch_lightning/trainer/progress.py:154]: There's no need for this. Loops customization is not public yet

File: pytorch_lightning/trainer/progress.py (hunk 2 above)
Comment (carmocca) [reply pytorch_lightning/trainer/progress.py:194]: It runs during `Loop.run` aka "on run"

**Pull request merged**